File: cpp/program/family.cpp
#ifndef FAMILY_CPP
#define FAMILY_CPP

#include <bits/stdc++.h>
#include "mobil.cpp"
using namespace std;

class Family : public Mobil{
    private:
        string kapasitasBagasi;
        string fiturKeselamatan;
        string fiturKeluarga;
    public:
        Family() : Mobil(){
            kapasitasBagasi = "";
            fiturKeselamatan = "";
            fiturKeluarga = "";
        }
        
        Family(string nama, string merk, Mesin mesin, int seat, string warna, int tahun, string harga, string kapasitasBagasi, string fiturKeselamatan, string fiturKeluarga) : Mobil(nama, merk, mesin, seat, warna, tahun, harga){
            this->kapasitasBagasi = kapasitasBagasi;
            this->fiturKeselamatan = fiturKeselamatan;
            this->fiturKeluarga = fiturKeluarga;
        }

        string getKapasitasBagasi(){
            return kapasitasBagasi;
        }

        string getFiturKeselamatan(){
            return fiturKeselamatan;
        }

        string getFiturKeluarga(){
            return fiturKeluarga;
        }

        void setKapasitasBagasi(string kapasitasBagasi){
            this->kapasitasBagasi = kapasitasBagasi;
        }

        void setFiturKeselamatan(string fiturKeselamatan){
            this->fiturKeselamatan = fiturKeselamatan;
        }

        void setFiturKeluarga(string fiturKeluarga){
            this->fiturKeluarga = fiturKeluarga;
        }

        void tampilkanInfo() const override {
            Mobil::tampilkanInfo();
            cout << "Kapasitas Bagasi: " << kapasitasBagasi << endl;
            cout << "Fitur Keselamatan: " << fiturKeselamatan << endl;
            cout << "Fitur Keluarga: " << fiturKeluarga << endl;
        }

        ~Family(){
            // Destructor
        }
};

#endif

File: java/program/Family.java
class Family extends Mobil{
    private String  kapasitasBagasi;
    private String fiturKeselamatan;
    private String fiturKeluarga;

    public Family() {
        super();
        this.kapasitasBagasi = "";
        this.fiturKeselamatan = "";
        this.fiturKeluarga = "";
    }

    public Family(String nama, String merk, Mesin mesin, int seat, String warna, int tahun, String harga, String kapasitasBagasi, String fiturKeselamatan, String fiturKeluarga) {
        super(nama, merk, mesin, seat, warna, tahun, harga);
        this.kapasitasBagasi = kapasitasBagasi;
        this.fiturKeselamatan = fiturKeselamatan;
        this.fiturKeluarga = fiturKeluarga;
    }

    public String getKapasitasBagasi() {
        return kapasitasBagasi;
    }

    public String getFiturKeselamatan() {
        return fiturKeselamatan;
    }

    public String getFiturKeluarga() {
        return fiturKeluarga;
    }

    public void setKapasitasBagasi(String kapasitasBagasi) {
        this.kapasitasBagasi = kapasitasBagasi;
    }

    public void setFiturKeselamatan(String fiturKeselamatan) {
        this.fiturKeselamatan = fiturKeselamatan;
    }

    public void setFiturKeluarga(String fiturKeluarga) {
        this.fiturKeluarga = fiturKeluarga;
    }

    @Override
    public void tampilkanInfo() {
        super.tampilkanInfo();
        System.out.println("Kapasitas Bagasi: " + kapasitasBagasi);
        System.out.println("Fitur Keselamatan: " + fiturKeselamatan);
        System.out.println("Fitur Keluarga: " + fiturKeluarga);
    }
}